rxbinding/src/main/java/com/jakewharton/rxbinding2/InitialValueObservable.java

```diff
@@ -0,0 +1,37 @@
+package com.jakewharton.rxbinding2;
+
+import android.support.annotation.MainThread;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+
+import static com.jakewharton.rxbinding2.internal.Preconditions.checkMainThread;
+
+public abstract class InitialValueObservable<T> extends Observable<T> {
+  @Override protected final void subscribeActual(Observer<? super T> observer) {
+    if (checkMainThread(observer)) {
```

Comment:
I moved it before I made this a public type since it's a requirement of _all_ subclasses in the library. It probably makes sense to leave the check inside the subclass `subscribeListener` method though.